src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,132 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.RegistrationException;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static RegistrationService registrationService;
+    private static final String SHORT_PASSWORD = "pswrd";
+    private static final String SHORT_LOGIN = "login";
+    private static final String MIN_LENGTH_PASSWORD = "paswrd";
+    private static final String MIN_LENGTH_LOGIN = "newlog";
+    private static final int BELOW_MIN_AGE = 17;
+    private static final int MIN_AGE = 18;
+    private static final String VALID_PASSWORD = "TheStrongestPassword";
+    private static final String VALID_LOGIN = "NewUser";
+    private static final int VALID_AGE = 30;
+    private static final String EMPTY_STRING = "";
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void invalidPassword_NotOk() {
+        user = new User(VALID_LOGIN, SHORT_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void invalidLogin_NotOk() {
+        user = new User(SHORT_LOGIN, VALID_PASSWORD, VALID_AGE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void theAgeIsLowerThanEighteen_NotOk() {
```

Comment:
change methods names in this class
this should be `register_underage_notOk`
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/69c2a03b-7fe0-4381-8940-90000e6a9f56)